torch/_dynamo/symbolic_convert.py

```diff
@@ -4164,6 +4164,25 @@ def speculate(self) -> SpeculationEntry:
             self.instructions[self.instruction_pointer - 1],
         )
 
+    def _get_frame_loc_chain(
+        self, frame_loc: tuple[str, int]
+    ) -> tuple[tuple[str, int], ...]:
+        frame_loc_chain: tuple[tuple[str, int], ...] = (frame_loc,)
+
+        if config.nested_graph_breaks:
+            current_tx: Optional[InstructionTranslatorBase] = self.parent
+            while current_tx is not None:
+                parent_frame_loc = (
+                    current_tx.f_code.co_filename,
+                    current_tx.lineno
+                    if current_tx.lineno >= 0
+                    else current_tx.f_code.co_firstlineno,
+                )
+                frame_loc_chain = (parent_frame_loc,) + frame_loc_chain
```

Comment:
Minor nit - I think repeated tuple addition is inefficient - appending to a list and converting to tuple afterward is the pattern I see more